ironfish/src/rpc/routes/chain/importChain.ts

```diff
@@ -0,0 +1,75 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+import bufio from 'bufio'
+import * as yup from 'yup'
+import { Assert } from '../../../assert'
+import { readBlock } from '../../../network/utils/block'
+import { IronfishNode } from '../../../node'
+import { Block } from '../../../primitives'
+import { ApiNamespace, router } from '../router'
+
+export type ImportSnapshotRequest =
+  | {
+      chunkStart?: number | null
+      chunkEnd?: number | null
+      blocks?: Buffer | null
+    }
+  | undefined
+
+export type ImportSnapshotResponse = {
+  headSeq: number
+}
+
+export const ImportSnapshotRequestSchema: yup.ObjectSchema<ImportSnapshotRequest> = yup
+  .object({
+    chunkStart: yup.number().nullable().optional(),
+    chunkEnd: yup.number().nullable().optional(),
+    blocks: yup.mixed<Buffer>().nullable().optional(),
+  })
+  .optional()
+
+export const ImportSnapshotSchema: yup.ObjectSchema<ImportSnapshotResponse> = yup
+  .object({
+    headSeq: yup.number().defined(),
+  })
+  .defined()
+
+router.register<typeof ImportSnapshotRequestSchema, ImportSnapshotResponse>(
+  `${ApiNamespace.chain}/importSnapshot`,
+  ImportSnapshotRequestSchema,
+  async (request, node): Promise<void> => {
+    Assert.isNotNull(node.chain.head, 'head')
+    Assert.isNotNull(node.chain.latest, 'latest')
+
+    if (!request.data?.blocks) {
+      request.end({ headSeq: node.chain.head.sequence })
+    }
+
+    const blocks = request.data?.blocks
+    if (blocks) {
+      const reader = bufio.read(blocks, true)
+      const deserializedBlocks = deserializeChunk(node, reader)
+
+      for (const block of deserializedBlocks) {
+        await node.chain.addBlock(block)
```

Comment:
We should handle the return value of `addBlock` -- if it returns `added: false` with an error, we should log that out. It'd be nice to propagate the error back through to the CLI command, but I think that could be a follow-up.

Reply:
Added. I chose to log through the `logger` object on `node` in line with other RPC routes.